### PR TITLE
Toggle3d

### PR DESCRIPTION
**Pull request by TheCovenant**

### CODE REVIEW

File: app/geo/gui/src/ts/twxplore/gui/geo/components/ToggleLayerPanel/ToggleLayerPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      container
+      alignItems={"center"}
+      spacing={2}
+      wrap="wrap"
+      direction="column"
+      xs
+    >
+      {Object.values(FeatureType).map(featureType => {
+        const featureTypeStrategy = getFeatureTypeStrategyByName(featureType);
+        if (featureTypeStrategy.layerTypeToggleable) {
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/ToggleLayerPanel/ToggleLayerPanel.tsx:37]: if (!...) { return <React.Fragment/>; }

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 2]
```diff
@@ -175,6 +176,21 @@ const MapLayoutImpl: React.FunctionComponent = () => {
...
+
+          <Grid
+            container
+            item
+            spacing={0}
+            direction="column"
+            className={classes.drawerGridContainer}
+            justify={"flex-start"}
+          >
+            <Grid>
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:188]: Grid item

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 3]
```diff
@@ -175,6 +176,21 @@ const MapLayoutImpl: React.FunctionComponent = () => {
...
+              <Typography align={"center"}>Toggle Layer Type</Typography>
+            </Grid>
+
+            <ToggleLayerPanel />
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:192]: Needs to be surrounded with a Grid item?

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 2 above)
Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:188]: This is a grid item (and container). It's both. Which seems to be fine by material-ui standards.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 3 above)
Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:192]: ToggleLayerPanel returns a Grid item, so I don't think so.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 2 above)
Comment (gordom6) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:188]: My comment was on line 188, the inner <Grid>.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 3 above)
Comment (gordom6) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:192]: Usually better for components not to contribute to an "outer" context, in this case a Grid.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 2 above)
Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:188]: Oh, I see it now

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 3 above)
Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:192]: Makes sense. I was doing that for my other panels so I'll see if I can fix that right now.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 4]
```diff
@@ -151,15 +151,17 @@ const MapLayoutImpl: React.FunctionComponent = () => {
           <Grid
             container
             item
+            direction="column"
             spacing={0}
             justify={"flex-start"}
             className={classes.drawerGridContainer}
           >
-            <Grid item xs={12}>
+            <Grid item xs>
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:159]: You don't really need the xs.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 5]
```diff
@@ -151,15 +151,17 @@ const MapLayoutImpl: React.FunctionComponent = () => {
...
               <Typography align={"center"}>Filters</Typography>
             </Grid>
-
-            <FilterPanel />
+            <Grid container item direction="column">
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:162]: Not sure why you need another container here. The FilterPanel should have its own Grid container if it needs one.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 6]
```diff
@@ -170,11 +172,13 @@ const MapLayoutImpl: React.FunctionComponent = () => {
             className={classes.drawerGridContainer}
             justify={"flex-start"}
           >
-            <Grid item xs={12}>
+            <Grid item xs>
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:175]: Same as above.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 7]
```diff
@@ -170,11 +172,13 @@ const MapLayoutImpl: React.FunctionComponent = () => {
...
               <Typography align={"center"}>Toggle Layer Visibility</Typography>
             </Grid>
 
-            <SelectionPanel />
+            <Grid item container direction="column">
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:179]: Same as above.

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 8]
```diff
@@ -185,11 +189,21 @@ const MapLayoutImpl: React.FunctionComponent = () => {
             className={classes.drawerGridContainer}
             justify={"flex-start"}
           >
-            <Grid>
+            <Grid item xs>
               <Typography align={"center"}>Toggle Layer Type</Typography>
             </Grid>
 
-            <ToggleLayerPanel />
+            <Grid
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:196]: Needs to be a container?

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx [hunk 9]
```diff
@@ -185,11 +189,21 @@ const MapLayoutImpl: React.FunctionComponent = () => {
             className={classes.drawerGridContainer}
             justify={"flex-start"}
           >
-            <Grid>
+            <Grid item xs>
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:192]: Need xs?

File: app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx (hunk 8 above)
Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:196]: Line 196: I think so. ToggleLayerPanel returns item-containers which are "rows", each row representing a FeatureType and its toggle button. 
This outside item-container is to stack those rows as columns

Comment (gordom6) [reply app/geo/gui/src/ts/twxplore/gui/geo/components/map/MapLayout.tsx:196]: The outer Grid shouldn't know anything about what <ToggleLayerPanel> is doing. Components should be self-contained.

**Pull request merged**